specs/builder.go

```diff
@@ -0,0 +1,216 @@
+	}
+	prefixes = map[string]prefix.Prefixer{
+		"ascii.fixed":  prefix.ASCII.Fixed,
+		"ascii.l":      prefix.ASCII.L,
+		"ascii.ll":     prefix.ASCII.LL,
+		"ascii.lll":    prefix.ASCII.LLL,
+		"ascii.llll":   prefix.ASCII.LLLL,
+		"bcd.fixed":    prefix.BCD.Fixed,
+		"bcd.l":        prefix.BCD.L,
+		"bcd.ll":       prefix.BCD.LL,
+		"bcd.lll":      prefix.BCD.LLL,
+		"bcd.llll":     prefix.BCD.LLLL,
+		"hex.fixed":    prefix.Hex.Fixed,
+		"hex.l":        prefix.Hex.L,
+		"hex.ll":       prefix.Hex.LL,
+		"hex.lll":      prefix.Hex.LLL,
+		"hex.llll":     prefix.Hex.LLLL,
+		"ebcdic.fixed": prefix.EBCDIC.Fixed,
+		"ebcdic.l":     prefix.EBCDIC.L,
+		"ebcdic.ll":    prefix.EBCDIC.LL,
+		"ebcdic.lll":   prefix.EBCDIC.LLL,
+		"ebcdic.llll":  prefix.EBCDIC.LLLL,
+		"binary.fixed": prefix.Binary.Fixed,
+	}
+	encodings = map[string]encoding.Encoder{
+		"asciiEncoder":  encoding.ASCII,
+		"bcdEncoder":    encoding.BCD,
+		"ebcdicEncoder": encoding.EBCDIC,
+		"binaryEncoder": encoding.Binary,
+		"hexEncoder":    encoding.Hex,
+		"lBCDEncoder":   encoding.LBCD,
+	}
+)
+
+var Builder MessageSpecBuilder = &messageSpecBuilder{}
+
+type MessageSpecBuilder interface {
+	ImportJSON([]byte) (*iso8583.MessageSpec, error)
+	ExportJSON(spec *iso8583.MessageSpec) ([]byte, error)
+}
+
+type messageSpecBuilder struct{}
+
+type specDummy struct {
+	Name   string                `json:"name,omitempty" xml:"name,omitempty"`
+	Fields map[string]fieldDummy `json:"fields,omitempty" xml:"fields,omitempty"`
+}
+
+type fieldDummy struct {
+	Type        string    `json:"type,omitempty" xml:"type,omitempty"`
+	Length      int       `json:"length,omitempty" xml:"length,omitempty"`
+	IDLength    int       `json:"id_length,omitempty" xml:"id_length,omitempty"`
+	Description string    `json:"description,omitempty" xml:"description,omitempty"`
+	Enc         string    `json:"enc,omitempty" xml:"enc,omitempty"`
+	Prefix      string    `json:"prefix,omitempty" xml:"prefix,omitempty"`
+	Padding     *padDummy `json:"padding,omitempty" xml:"padding,omitempty"`
+}
+
+type padDummy struct {
+	Type string `json:"type" xml:"type"`
+	Pad  rune   `json:"pad" xml:"pad"`
+}
+
+func (builder *messageSpecBuilder) ImportJSON(raw []byte) (*iso8583.MessageSpec, error) {
+	dummy := specDummy{}
+	err := json.Unmarshal(raw, &dummy)
+	if err != nil {
+		return nil, err
+	}
+
+	if len(dummy.Fields) == 0 {
+		return nil, fmt.Errorf("invalid json spec file")
+	}
+
+	spec := iso8583.MessageSpec{
+		Name:   dummy.Name,
+		Fields: make(map[int]field.Field),
+	}
+
+	for key, dummyField := range dummy.Fields {
+		index := 0
+		_, err := fmt.Sscanf(key, FieldPrefix+"%d", &index)
+		if err != nil {
+			return nil, fmt.Errorf("invalid field index, index's format is `Field`+index ")
+		}
+
+		constructor := fieldConstructor[dummyField.Type]
+		if constructor == nil {
+			return nil, fmt.Errorf("unable create field with %s", dummyField.Type)
+		}
+
+		enc := encodings[dummyField.Enc]
```

Comment:
It would be great to check that value was found and if not, return the error message. We want developers know that encoding (prefixer, padder, etc.) with the provided name was not found.